### ISSUE TITLE
wake sandbox, option to 'disable network'

### ISSUE THREAD

**Issue by mmjconolly** [sifive/wake#450]
It'd be nice to be able to control if networking is available for a running job,
```
$ wake -x 'makePlan ("ip", "addr", Nil) Nil | tryDisableNetwork | runJob | getJobStdout'
1: lo: <LOOPBACK> mtu 65536 qdisc noop state DOWN group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
```

I'm imagining that `fuse-wake` could implement this in a similar way to how `unshare -r -n ip addr` would run.

Perhaps in a future version we could make it the default and have `enableNetworking` as the option

**Comment by terpstra** [comment]
For my clarification: you've implemented `tryDisableNetwork` or are asking to have it implemented?

If the latter, it should indeed be quite simple to add this. I guess we just extend the JSON fed to fuse-wake to indicate if a job should add the appropriate flag to unshare or not.

**Comment by mmjconolly** [comment]
I haven't implemented it, just writing it down as something that would be nice to have

**Comment by terpstra** [comment]
So, I would like to implement this. Technically, it's very easy to do. The tricky part is the API.

I still want a completely redesigned job environment API. However, in the interim, I could easily make this work by having fuse-wake look for a `"block-network"` String in the resources list. This is, of course, super hacky. OTOH, it would let us experiment with the feature without having to commit to redesigning the API right now.

**Comment by mmjconolly** [comment]
> I still want a completely redesigned job environment API. However, in the interim, I could easily make this work by having fuse-wake look for a `"block-network"` 

How about if we flip it around?
I'd like it if we had the option to globally say 'disallow network' (via whatever mechanism) and then in particular rules add a `allowNetworking`, `allowIPC` etc

**Comment by terpstra** [comment]
For a new API, I totally support that idea. I was more thinking about backwards compatibility.

**Comment by terpstra** [comment]
Fixed by #463
